redwood-treehouse-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/treehouse/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -40,50 +50,76 @@ internal class UIViewLazyList<A : AppService>(
   treehouseApp: TreehouseApp<A>,
   widgetSystem: WidgetSystem,
 ) : LazyList<UIView> {
-  private val dataSource = TableViewDataSource(treehouseApp, widgetSystem)
-  private val root = UITableView().apply {
-    this.dataSource = this@UIViewLazyList.dataSource
-  }
+  private val layout = UICollectionViewFlowLayout()
+  private val viewController = CollectionViewController(treehouseApp, widgetSystem, layout)
 
   override fun isVertical(isVertical: Boolean) {
-    if (!isVertical) {
-      // TODO UITableView only supports vertical scrolling. Switch to UICollectionView.
-      TODO()
+    layout.scrollDirection = if (isVertical) {
+      UICollectionViewScrollDirectionVertical
+    } else {
+      UICollectionViewScrollDirectionHorizontal
     }
   }
 
   override fun intervals(intervals: List<LazyListInterval>) {
-    dataSource.intervals = intervals
-    root.reloadData()
+    viewController.dataSource.intervals = intervals
+    viewController.collectionView.reloadData()
   }
 
   override var layoutModifiers: LayoutModifier = LayoutModifier
 
-  override val value: UIView get() = root
+  override val value: UIView get() = viewController.collectionView
+}
+
+private class CollectionViewController<A : AppService>(
+  treehouseApp: TreehouseApp<A>,
+  widgetSystem: WidgetSystem,
+  layout: UICollectionViewLayout,
+) : UICollectionViewController(layout),
+  UICollectionViewDelegateFlowLayoutProtocol {
+  val dataSource = CollectionViewDataSource(treehouseApp, widgetSystem)
+
+  init {
+    collectionView.registerClass(CollectionViewCell, "CollectionViewCell")
+    collectionView.dataSource = dataSource
+  }
+
+  override fun collectionView(collectionView: UICollectionView, layout: UICollectionViewLayout, sizeForItemAtIndexPath: NSIndexPath): CValue<CGSize> {
+    // TODO Flip the dimensions when isVertical==false.
+    return CGSizeMake(
+//      width = collectionView.bounds.useContents { size }.width, // This always returns 0.
+      width = 100.0,
+      height = 64.0,
+    )
+  }
```

Comment:
The motivation here is to make sure that we have a single column (for `LazyColumn`) or a single row (for `LazyRow`). Without this, the cells are laid out in a grid pattern.

I attempted to set the width to the `collectionView` width, but that always seems to return 0. Any ideas what I'm doing wrong? Is there a better approach to achieving this than what I have here?

Reply:
Sounds like the `collectionView` hasn't been sized yet, so you might want to call [sizeThatFits](https://developer.apple.com/documentation/uikit/uiview/1622625-sizethatfits) and return the width of that `CGSize`.